packages/next/client/page-loader.js

```diff
@@ -132,6 +132,14 @@ export default class PageLoader {
     }
     this.prefetchCache.add(scriptRoute)
 
+    // Inspired by quicklink, license: https://github.com/GoogleChromeLabs/quicklink/blob/master/LICENSE
+    // Don't prefetch if the user is on 2G / Don't prefetch if Save-Data is enabled
+    if ('connection' in navigator) {
```

Comment:
Interestingly including this condition on the `if` below wouldn't make a difference since Terser optimizes this, so keeping it separate for readability.